test/parallel/test-assert-objects.js

```diff
@@ -0,0 +1,368 @@
+    const obj1 = { fn: () => {} };
+    const obj2 = { fn: () => {} };
+    assert.throws(() => assert.matchObjectStrict(obj1, obj2), Error);
+  });
+
+  it('should loosely compare two objects with undefined and missing properties', function() {
+    const obj1 = { a: undefined };
+    const obj2 = {};
+    assert.throws(() => assert.matchObject(obj1, obj2), Error);
+  });
+
+  it('should strictly compare two objects with mixed types of properties', function() {
+    const sym = Symbol('test');
+    const obj1 = { num: 1, str: 'test', bool: true, sym: sym };
+    const obj2 = { num: 1, str: 'test', bool: true, sym: sym };
+    assert.matchObjectStrict(obj1, obj2);
+  });
+
+  it('should compare two objects with Buffers', function() {
+    const obj1 = { buf: Buffer.from('Node.js') };
+    const obj2 = { buf: Buffer.from('Node.js') };
+    assert.matchObjectStrict(obj1, obj2);
+    assert.matchObject(obj1, obj2);
+  });
+
+  it('should strictly compare two objects with identical Error properties', function() {
+    const error = new Error('Test error');
+    const obj1 = { error: error };
+    const obj2 = { error: error };
+    assert.matchObjectStrict(obj1, obj2);
+  });
+
+  it('should not strictly compare two objects with different Error instances', function() {
+    const obj1 = { error: new Error('Test error 1') };
+    const obj2 = { error: new Error('Test error 2') };
+    assert.throws(() => assert.matchObjectStrict(obj1, obj2), Error);
+  });
+
+  it('should strictly compare two objects with identical Typed Array properties', function() {
+    const typedArray = new Uint8Array([1, 2, 3]);
+    const obj1 = { typedArray: typedArray };
+    const obj2 = { typedArray: typedArray };
+    assert.matchObjectStrict(obj1, obj2);
+  });
+
+  it('should not strictly compare two objects with different Typed Array instances', function() {
+    const obj1 = { typedArray: new Uint8Array([1, 2, 3]) };
+    const obj2 = { typedArray: new Uint8Array([4, 5, 6]) };
+    assert.throws(() => assert.matchObjectStrict(obj1, obj2), Error);
+  });
+
+  it('should strictly compare two Map objects with identical entries', function() {
+    const map1 = new Map([
+      ['key1', 'value1'],
+      ['key2', 'value2'],
+    ]);
+    const map2 = new Map([
+      ['key1', 'value1'],
+      ['key2', 'value2'],
+    ]);
+    assert.matchObjectStrict(map1, map2);
+  });
+
+  it('should not strictly compare two Map objects with different entries', function() {
+    const map1 = new Map([
+      ['key1', 'value1'],
+      ['key2', 'value2'],
+    ]);
+    const map2 = new Map([
+      ['key1', 'value1'],
+      ['key3', 'value3'],
+    ]);
+    assert.throws(() => assert.matchObjectStrict(map1, map2), Error);
+  });
+
+  it('should strictly compare two Set objects with identical values', function() {
+    const set1 = new Set(['value1', 'value2']);
+    const set2 = new Set(['value1', 'value2']);
+    assert.matchObjectStrict(set1, set2);
+  });
```

Comment:
Can we add tests with `Map` and `Set` coming from different realm? Currently, here's what I'm getting:

```console
$ ./node -e 'assert.matchObjectStrict(new Set, new vm.runInNewContext("Set"))'
node:assert:131
  throw new AssertionError(obj);
  ^

AssertionError [ERR_ASSERTION]: Set(0) {} matchObjectStrict [Function: Set]
    at [eval]:1:8
    at runScriptInThisContext (node:internal/vm:209:10)
    at node:internal/process/execution:118:14
    at [eval]-wrapper:6:24
    at runScript (node:internal/process/execution:101:62)
    at evalScript (node:internal/process/execution:136:3)
    at node:internal/main/eval_string:49:3 {
  generatedMessage: true,
  code: 'ERR_ASSERTION',
  actual: Set(0) {},
  expected: [Function: Set],
  operator: 'matchObjectStrict'
}

Node.js v23.0.0-pre
```

Reply:
Added the tests and added support for different realms `Set` and `Map`